src/Tasks/ManifestUtil/CngLightup.cs

```diff
@@ -115,30 +111,6 @@ internal static RSA GetRSAPrivateKey(X509Certificate2 cert)
             return s_getRsaPrivateKey(cert);
         }
 
-        internal static DSA GetDSAPublicKey(X509Certificate2 cert)
```

Comment:
It looks like this is called via reflection, and I don't think it should be removed. I think we should not enable this rule, because I don't understand the compatibility implications of removing support for this style of signing at build time.

https://github.com/dotnet/msbuild/blob/1fea1d30670e0ae06ca8d2a1ddeab8d3c23ec59d/src/Tasks/ManifestUtil/CngLightup.cs#L520-L530

Reply:
Ahh bummer, I thought we could get away with it because it wasn't referenced anywhere and it was not a public API. 

The code you linked is a private function, it is called via GetDSAPublicKey/GetDSAPrivateKey:
https://github.com/dotnet/msbuild/blob/14efa06d736aabb3af01ff827fff9426518f5fac/src/Tasks/ManifestUtil/CngLightup.cs#L118-L140

The above methods call BindCoreDelegate is using reflection to call System.Security.Cryptography.X509Certificates.**DSACertificateExtensions**.GetDSAPublicKey/GetDSAPrivateKey

Additionally, CngLightup is only used when `RUNTIME_TYPE_NETCORE=false`.

The only public exposed functions I see that end up calling CNGLightup are in src/Tasks/ManifestUtil/SecurityUtil.cs

All calls end up in:
https://github.com/dotnet/msbuild/blob/14efa06d736aabb3af01ff827fff9426518f5fac/src/Tasks/ManifestUtil/SecurityUtil.cs#L599

If the file being signed is a portable executable it will use `signtool.exe` via `SignPEFileInternal`, if it isn't it will then call GetRSAPrivateKey and throw an `OnlyRSACertsAreAllowed` error if its null.

https://github.com/dotnet/msbuild/blob/14efa06d736aabb3af01ff827fff9426518f5fac/src/Tasks/ManifestUtil/SecurityUtil.cs#L625-L637

There is then the file header comment, which I think might mean this file is pulled in externally somewhere?:
https://github.com/dotnet/msbuild/blob/518c041f4511a6bc23eb40703b69a94ea46c65fd/src/Tasks/ManifestUtil/CngLightup.cs#L14-L17